platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -76,19 +73,16 @@ public List<NodeId> getNeighbors() {
      */
     @Override
     public List<NodeId> getNeighbors(final Predicate<NodeId> filter) {
```

Comment:
This method used to return *neighbors*, but now it's returning all peers.

Reply:
the peers are all neighbors, but not necessarily adjacent.

Reply:
I don't understand this. If all peers are neighbors, why do we have 2 terms here? To me, `neighbor` implies `adjacent`

Reply:
Ok, I think I understand why this works.

If the node ID of the neighbor is greater than ours, then this method returns a value that is one less than its actual index. If the node ID of the neighbor is equal to our node ID plus one, `getIndexOfNodeId()` increments the number so that it doesn't actually equal our index.

Although this technically yields the correct result, I feel like this is one of those times where the code is "accidentally returning the right results". At the very least, the code here does not return values consistent with the javadocs. Better to refactor this code, I can imagine that the reason why/how this works may not be apparent to a developer who looks at this in the future.

Reply:
resolving, as this is now outdated as per subsequent discussions.